pep-0678.rst

```diff
@@ -114,36 +119,39 @@ includes a note of the minimal failing example::
 
 Non-goals
 ---------
-``__note__`` is *not* intended to carry structured data.  If your note is for
-use by a program rather than display to a human, `we recommend
+Tracking multiple notes as a tuple, rather than by concatenating strings when
+notes are added, is intended to maintain the distinction between strings
+which might be translated by packages such as ``friendly-traceback``.
+
+However, ``__notes__`` is *not* intended to carry structured data.  If your
+note is for use by a program rather than display to a human, `we recommend
 <https://discuss.python.org/t/accepting-pep-654-exception-groups-and-except/10813/26>`__
 instead (or additionally) choosing a convention for an attribute, e.g.
 ``err._parse_errors = ...`` on the error or ``ExceptionGroup``.
 
-As a rule of thumb, prefer `exception chaining
+As a rule of thumb, we suggest that you should prefer `exception chaining
 <https://docs.python.org/3/tutorial/errors.html#exception-chaining>`__ when the
 error is going to be re-raised or handled as an individual error, and prefer
-``__note__`` when you are collecting multiple exception objects to handle
+``.add_note()`` when you are collecting multiple exception objects to handle
 together or later. [1]_
 
 
 Specification
 =============
 
-``BaseException`` gains a new mutable attribute ``__note__``, which defaults to
-``None`` and may have a string assigned.  When an exception with a note is
-displayed, the note is displayed immediately after the exception.
-
-Assigning a new string value overrides an existing note; if concatenation is
-desired users are responsible for implementing it with e.g.::
-
-    e.__note__ = msg if e.__note__ is None else e.__note__ + "\n" + msg
+``BaseException`` gains a new read-only field ``__notes__``, an initially empty
+tuple of strings, and a new method ``.add_note(note: str | None, *, replace:
+bool=False)``. If ``note`` is not ``None``, it is added to the exception's
+notes which appear in the standard traceback after the exception string. If
+``replace`` is true, all previously existing notes are removed before the new
+one is added. To clear all notes, use ``add_note(None, replace=True)``. A
+:exc:`TypeError` is raise if ``note`` is neither a string nor ``None``.
```

Comment:
```suggestion
``TypeError`` is raised if ``note`` is neither a string nor ``None``.
```

Fix error in legacy docutils build; docutils doesn't support the Sphinx-only `:exc:` directive (and it doesn't really make a difference here anyway unless we format it specially in CSS).

Also, fix grammar.